python/tests/test_joining.py

```diff
@@ -0,0 +1,229 @@
+from __future__ import print_function
+import ctypes
+
+import pytest
+
+import numpy as np
+from numba import cuda
+
+from libgdf_cffi import ffi, libgdf
+
+from .utils import new_column, unwrap_devary, get_dtype
+
+
+@pytest.mark.parametrize('dtype', [np.int32, np.int64, np.float32, np.float64])
+def test_innerjoin(dtype):
+    cuda.current_context()
+    # Make data
+    left = np.array([0, 0, 1, 2, 3], dtype=dtype)
+    d_left = cuda.to_device(left)
```

Comment:
There is a lot of duplicated code in the tests to make the left and right GDFs from the input arrays.  Could this be extracted into some helper functions to make the test cases more readable?